lib/Dialect/SPIRV/SPIRVOps.cpp

```diff
@@ -199,6 +199,162 @@ static void printNoIOOp(Operation *op, OpAsmPrinter *printer) {
   printer->printOptionalAttrDict(op->getAttrs());
 }
 
+//===----------------------------------------------------------------------===//
+// spv.AccessChainOp
+//===----------------------------------------------------------------------===//
+
+// Function to extract an actual value from ConstantOp.
+static std::pair<StringRef, bool> extractValueFromConstOp(Operation *op,
+                                                          int32_t &indexValue) {
+  // spv.constant operation contains kValueAttrName.
+  auto valueAttr = op->getAttr(kValueAttrName);
+  if (!valueAttr) {
+    return {
+        "index must be an integer constant to access an element of StructType",
+        false};
+  }
+
+  auto integerValueAttr = valueAttr.dyn_cast<IntegerAttr>();
+  if (!integerValueAttr) {
+    return {"expected an integer constant for index", false};
+  }
+
+  indexValue = integerValueAttr.getInt();
+  return {"", true};
+}
+
+static ParseResult parseAccessChainOp(OpAsmParser *parser,
+                                      OperationState *state) {
+  OpAsmParser::OperandType ptrInfo;
+  SmallVector<OpAsmParser::OperandType, 4> indicesInfo;
+  spirv::StorageClass resultStorageClass;
+  Type type;
+  llvm::SMLoc indicesLocation;
+  int32_t index;
+  Type indicesType = parser->getBuilder().getIndexType();
+
+  if (parser->parseOperand(ptrInfo) ||
+      parser->getCurrentLocation(&indicesLocation) ||
+      parser->parseOperandList(indicesInfo, OpAsmParser::Delimiter::Square) ||
+      parser->parseColonType(type) ||
+      parser->resolveOperand(ptrInfo, type, state->operands) ||
+      parser->resolveOperands(indicesInfo, indicesType, state->operands)) {
+    return failure();
+  }
+
+  if (!indicesInfo.size()) {
+    return parser->emitError(indicesLocation,
+                             "expected at least one index for spv.AccessChain");
+  }
+
+  auto ptrType = type.dyn_cast<spirv::PointerType>();
+  if (!ptrType) {
+    return parser->emitError(
+               indicesLocation,
+               "expected a pointer to composite type, but provided ")
+           << type;
+  }
+
+  resultStorageClass = ptrType.getStorageClass();
+  auto resultType = ptrType.getPointeeType();
+
+  for (uint64_t i = 0; i < indicesInfo.size(); ++i) {
+    if (auto cType = resultType.dyn_cast<spirv::CompositeType>()) {
+      index = 0;
+      if (resultType.isa<spirv::StructType>()) {
+        Operation *op = state->operands[i + 1]->getDefiningOp();
```

Comment:
This should likely be left to the verifier, there isn't a guarantee that this refers to what you expect when the IR is being parsed.

Reply:
Thanks for review, in case of AccessChain the index type for the !svp.struct must be an spv.constant, so, I hope it's a valid operation to get its value, in case of other composite type, we just getting the element type. It seems like the only way to get the resultType is walking into composite type by indices.  It was some small discussion in other tread, but I might miss something, please fix me if I'm wrong.
Thanks.